test/native/helpers.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * External dependencies
+ */
+import { act, render, fireEvent } from '@testing-library/react-native';
+
+// Set up the mocks for getting the HTML output of the editor
+let triggerHtmlSerialization = () => {};
+let serializedHtml;
+const bridgeMock = jest.requireMock( '@wordpress/react-native-bridge' );
+bridgeMock.subscribeParentGetHtml = jest.fn( ( callback ) => {
+	triggerHtmlSerialization = callback;
+} );
+bridgeMock.provideToNative_Html = jest.fn( ( html ) => {
+	serializedHtml = html;
+} );
+
+export async function initializeEditor( { initialHtml } ) {
+	const Editor = require( '@wordpress/edit-post/src/editor' ).default;
+	const renderResult = render(
+		<Editor
+			postId={ 0 }
+			postType="post"
+			initialTitle="test"
+			initialHtml={ initialHtml }
+		/>
+	);
+	const { getByTestId } = renderResult;
+
+	const blockListWrapper = await waitFor( () =>
+		getByTestId( 'block-list-wrapper' )
+	);
+
+	// onLayout event has to be explicitly dispatched in BlockList component,
+	// otherwise the inner blocks are not rendered.
+	fireEvent( blockListWrapper, 'layout', {
+		nativeEvent: {
+			layout: {
+				width: 100,
+			},
+		},
+	} );
+
+	return renderResult;
+}
+
+export * from '@testing-library/react-native';
+
+// Custom implementation of the waitFor utility to prevent the issue: https://git.io/JYYGE
+export function waitFor( cb, timeout = 150 ) {
```

Comment:
After checking the issue described [here](https://git.io/JYYGE), I noticed that it mostly happens when you have more than one `await` call to a function that internally uses `act` (reference). In my case, having more than one call to any of the following functions caused the test to fail with the error:
`Warning: You called act(async () => ...) without await. This could lead to unexpected testing behaviour, interleaving multiple act calls and mixing their scopes. You should - await act(async () => ...);`

**Functions affected:**
- `waitFor`
- `queryBy*`
- `queryAllBy*`
- `findBy*`
- `findAllBy*`

This is a critical blocker because most of our integration tests would require waiting for more than one update 😞 . As a workaround, I decided to implement our custom version of `waitFor` avoiding using `async/await`, I saw that the [original `waitFor` function](https://github.com/callstack/react-native-testing-library/blob/a018c7e1b14f9edc03940bc5f8f97303d95b271d/src/waitFor.js#L34) is more complex and covers more logic than this function but for now, I think it could help us to unblock the integration tests.

The idea is that we use this function every time we want to wait for something between DOM updates, I have some examples on the reusable block tests that can be handy to understand the use.

**Example for getting a block:**
```
const reusableBlock = await waitFor( () =>
	getByA11yLabel( /Reusable block Block\. Row 1/ )
);
```

Reply:
Sounds good. Thank you for exploring this and offering an alternative to unblock us.